templates/zookeeper-dns.yaml

```diff
@@ -0,0 +1,241 @@
+                          response.send(event, context, resource.FAILED, responseData);
+                      } else {
+                          responseData = { AutoScalingGroupName: data.EnvironmentResources.AutoScalingGroups[0].Name };
+                          response.send(event, context, response.SUCCESS, responseData);
+                      }
+                  });
+              } else {
+                  responseData = {Error: 'Environment name not specified'};
+                  console.log(responseData.Error);
+                  response.send(event, context, response.FAILED, responseData);
+              }
+          };
+      Handler: index.handler
+      Runtime: nodejs4.3
+      Timeout: '10'
+      VpcConfig:
+        SubnetIds: !Ref 'SubnetID'
+        SecurityGroupIds: !Ref 'SecurityGroups'
+      Role: !GetAtt [LambdaExecutionRole, Arn]
+  LambdaExecutionRole:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service: [lambda.amazonaws.com]
+            Action: ['sts:AssumeRole']
+      Path: /
+      ManagedPolicyArns:
+        - arn:aws:iam::aws:policy/service-role/AWSElasticBeanstalkEnhancedHealth
+        - arn:aws:iam::aws:policy/service-role/AWSLambdaVPCAccessExecutionRole
+        - arn:aws:iam::aws:policy/service-role/AWSLambdaENIManagementAccess
+      Policies:
+        - PolicyName: !Sub '${AWS::StackName}-zookeeper'
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+            - Effect: Allow
+              Action: ['logs:CreateLogGroup', 'logs:CreateLogStream', 'logs:PutLogEvents']
+              Resource: arn:aws:logs:*:*:*
+            - Effect: Allow
+              Action: ['cloudformation:DescribeStacks', 'elasticbeanstalk:Describe*']
+              Resource: '*'
+            - Effect: Allow
+              Action: ['route53:*']
+              Resource: ['*']
+  CloudWatchASGRule:
+    Type: AWS::Events::Rule
+    Properties:
+      EventPattern:
+        source:
+          - aws.autoscaling
+        detail-type:
+          - EC2 Instance Launch Successful
+          - EC2 Instance Terminate Successful
+        detail:
+          AutoScalingGroupName:
+            - !GetAtt [BeanstalkStack, AutoScalingGroupName]
+      State: ENABLED
+      Targets:
+        -
+          Arn: !GetAtt [UpsertRoute53Record, Arn]
+          Id: UpsertRoute53RecordV1
+  LambdaInvokePermission:
+    Type: AWS::Lambda::Permission
+    Properties:
+      FunctionName: !GetAtt [UpsertRoute53Record, Arn]
+      Action: lambda:InvokeFunction
+      Principal: events.amazonaws.com
+  InitializeUpsertLambda:
+    Type: Custom::InitializeUpsertLambda
+    Properties:
+      ServiceToken: !GetAtt [UpsertRoute53Record, Arn]
+      ZookeeperASGName: !GetAtt [BeanstalkStack, AutoScalingGroupName]
+  UpsertRoute53Record:
+    Type: AWS::Lambda::Function
```

Comment:
Would be helpful to have a comment here describing the purpose of this Lambda function